File: prompts/fixtures/python/debug_crash.py
import sys
import os
import signal


def divide(a, b):
    return a // b              # line 7 — will crash when b=0


def dereference_none():
    obj = None                 # line 11
    print(obj.value)           # line 12 — AttributeError


def abort_handler():
    os.kill(os.getpid(), signal.SIGABRT)  # line 16 — SIGABRT


def main():
    if len(sys.argv) < 2:                         # line 20
        print(f"Usage: {sys.argv[0]} [divzero|none|abort]")  # line 21
        return 1                                   # line 22
    mode = sys.argv[1]                             # line 23
    print(f"mode: {mode}")                         # line 24
    if mode.startswith("d"):                       # line 25
        divide(10, 0)                              # line 26
    elif mode.startswith("n"):                     # line 27
        dereference_none()                         # line 28
    elif mode.startswith("a"):                     # line 29
        abort_handler()                            # line 30
    return 0                                       # line 31


if __name__ == "__main__":
    sys.exit(main())

File: prompts/fixtures/cpp/debug_crash.cpp
#include <cstdio>
#include <cstdlib>
#include <cstring>

int divide(int a, int b) {
    return a / b;              // line 6 — will crash when b=0
}

void dereference_null() {
    int *p = nullptr;          // line 10
    std::printf("%d\n", *p);   // line 11 — SIGSEGV
}

void abort_handler() {
    std::abort();              // line 15 — SIGABRT
}

int main(int argc, char *argv[]) {
    if (argc < 2) {                                        // line 19
        std::printf("Usage: %s [divzero|null|abort]\n", argv[0]);  // line 20
        return 1;                                          // line 21
    }
    std::printf("mode: %s\n", argv[1]);                   // line 23
    if (argv[1][0] == 'd') {                              // line 24
        divide(10, 0);                                     // line 25
    } else if (argv[1][0] == 'n') {                       // line 26
        dereference_null();                                // line 27
    } else if (argv[1][0] == 'a') {                       // line 28
        abort_handler();                                   // line 29
    }
    return 0;                                              // line 31
}
